File: SoojinKim/2023-10-08-Non-overlapping-Intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals,(arr1,arr2) -> arr1[1] - arr2[1]);
        int answer = 0;
        int prev = intervals[0][1];
        for(int i=1; i<intervals.length; i++){
            if(prev > intervals[i][0]){
                answer++;
            }else{
                prev=intervals[i][1];
            }
        }
        
    return answer;
    }
}

File: MinguKim/2023-10-08-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        count = 1  
        intervals.sort(key=lambda x: x[1])
        compare = intervals[0][1]

        for i in range(1, len(intervals)):
            if intervals[i][0] >= compare:
                count += 1
                compare = intervals[i][1]
        if len(intervals) == 0:
            return 0

        return len(intervals) - count